server/stat.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"crypto/x509"
+
+	"github.com/pingcap/tidb/sessionctx/variable"
+	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/zap"
+)
+
+var (
+	serverNotAfter  = "Ssl_server_not_after"
+	serverNotBefore = "Ssl_server_not_before"
+)
+
+var defaultStatus = map[string]*variable.StatusVal{
+	serverNotAfter:  {Scope: variable.ScopeGlobal | variable.ScopeSession, Value: ""},
+	serverNotBefore: {Scope: variable.ScopeGlobal | variable.ScopeSession, Value: ""},
+}
+
+// GetScope gets the status variables scope.
+func (s *Server) GetScope(status string) variable.ScopeFlag {
+	return variable.DefaultStatusVarScopeFlag
+}
+
+// Stats returns the server statistics.
+func (s *Server) Stats(vars *variable.SessionVars) (map[string]interface{}, error) {
+	m := make(map[string]interface{}, len(defaultStatus))
+
+	for name, v := range defaultStatus {
+		m[name] = v.Value
+	}
+
+	tlsConfig := s.getTLSConfig()
+	if tlsConfig != nil {
+		if len(tlsConfig.Certificates) == 1 {
+			pc, err := x509.ParseCertificate(tlsConfig.Certificates[0].Certificate[0])
+			if err != nil {
+				logutil.BgLogger().Error("Failed to parse TLS certficates to get server status", zap.Error(err))
+			} else {
+				m[serverNotAfter] = pc.NotAfter.Format("Jan 2 15:04:05 2006 MST")
```

Comment:
```suggestion
				m[serverNotAfter] = pc.NotAfter.Format("Jan _2 15:04:05 2006 MST")
```
I just checked MySQL 8.0.25 and it shows:
```
May  5 08:48:56 2021 GMT
```
I.e. with an extra space between Month and Day, so I wonder if it should be space-padded?

MySQL seems to use the openssl function `ASN1_TIME_print` which resolves to `ossl_asn1_time_print_ex` with the format:
```c
527             return BIO_printf(bp, "%s %2d %02d:%02d:%02d%.*s %d%s",
528                           _asn1_mon[stm.tm_mon], stm.tm_mday, stm.tm_hour,
529                           stm.tm_min, stm.tm_sec, f_len, f, stm.tm_year + 1900,
530                           (gmt ? " GMT" : "")) > 0;
```